tools/mzxmlshaper/mzXMLShaper.xml

```diff
@@ -0,0 +1,152 @@
+<tool id="mzxmlshaper" name="Mz(X)ML Shaper" version="1.0.0+galaxy0" profile="21.01">
+  <description>
+    enables the conversion of open format files into XCMS-supported mz(X)ML file formats
+  </description>
+  <requirements>
+    <requirement type="package" version="4.3.3">r-base</requirement>
+    <requirement type="package" version="0.42.0">bioconductor-msdata</requirement>
+    <requirement type="package" version="2.36.0">bioconductor-mzr</requirement>
+    <requirement type="package" version="1.0.0">r-w4mrutils</requirement>
+  </requirements>
+  <command detect_errors="aggressive"><![CDATA[
+    Rscript '$__tool_directory__/mzXMLShaper.R'
+      inputfilename '$infile'
+      outputfileformat '$outformat'
+      #if str($outformat) == 'mzml':
+        outputfilename '$mzml_file'
+      #end if
+      #if str($outformat) == 'mzXml':
+        outputfilename '$mzXml_file'
+      #end if
+  ]]></command>
+  <inputs>
+    <param name="infile" type="data" format="mzml,mzxml,netcdf" label="File(s) to convert" help="file or collection"/>
+    <param name="outformat" type="select" label="Desired output format">
+      <option value="mzml">mzML</option>
+      <option value="mzXml">mzXML</option>
+    </param>
+  </inputs>
+  <outputs>
+    <data format="mzml" label="${infile.name.rsplit('.',1)[0]}.mzml" name="mzml_file">
```

Comment:
I would not do such surgery here. The names of Galaxy datasets can be changed anytime by the user. So its not clear you have a filetype ending in the name.

Reply:
Resolved